pennylane/ops/qubit.py

```diff
@@ -261,10 +278,16 @@ class CNOT(Operation):
     num_params = 0
     num_wires = 2
     par_domain = None
+    eigvals = np.array([1, 1, 1, -1])
```

Comment:
Just to clarify: eigenvalues are used in the case of an `Observable` and a `DiagonalOperation` for computations. These are defined for completeness i.e. users can query `CNOT.eigvals`, correct?

Reply:
That was the idea. But I start doubting if this is such a great idea, see comments above

Reply:
Right! My personal take would be that it's completely fine to only define `eigvals` in cases where they are internally used for now. The inherited `eigvals` property (provided that `_matrix` is defined for the class) should help out users if need be.

Reply:
That's what I'll do!